tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -55,6 +58,137 @@
 										  Int32GetDatum(0),                                        \
 										  Float8GetDatum(0)))
 
+/* TODO:: Q:: These should probably be in its own file? */
+Datum
+ts_integer_from_now_func_get_datum(int64 interval, Oid time_dim_type, Oid now_func)
+{
+	Datum now;
+
+	AssertArg(IS_INTEGER_TYPE(time_dim_type));
+
+	ts_integer_now_func_validate(now_func, time_dim_type);
+
+	// bgw luncher starts a transactions so here I can assume I am always inside a transaction
+	// right? no need for StartTransactionCommand();
+	/* functions in indexes may want a snapshot set */
+	// Q:: Why is there not one already?
+	PushActiveSnapshot(GetTransactionSnapshot());
+	now = OidFunctionCall0(now_func);
+	PopActiveSnapshot();
+
+	switch (time_dim_type)
+	{
+		case INT2OID:
+			return Int64GetDatum(DatumGetInt16(now) - interval);
+		case INT4OID:
+			return Int64GetDatum(DatumGetInt32(now) - interval);
+		case INT8OID:
+			return Int64GetDatum(DatumGetInt64(now) - interval);
+		default:
+			pg_unreachable();
+	}
+}
+
+void
+ts_integer_now_func_validate(Oid now_func_oid, Oid open_dim_type)
+{
+	HeapTuple tuple;
+	Form_pg_proc now_func;
+
+	/* this function should only be called for hypertabes with integer open time dimension */
+	Assert(IS_INTEGER_TYPE(open_dim_type));
+
+	if (!OidIsValid(now_func_oid))
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_FUNCTION), (errmsg("invalid chunk sizing function"))));
+
+	tuple = SearchSysCache1(PROCOID, ObjectIdGetDatum(now_func_oid));
+	if (!HeapTupleIsValid(tuple))
+		ereport(ERROR,
+				(errcode(ERRCODE_NO_DATA_FOUND),
+				 errmsg("cache lookup failed for function %u", now_func_oid)));
+
+	now_func = (Form_pg_proc) GETSTRUCT(tuple);
+
+	if ((now_func->provolatile != PROVOLATILE_IMMUTABLE &&
+		 now_func->provolatile != PROVOLATILE_STABLE) ||
+		now_func->pronargs != 0)
+		// q:: todo:: do I have to ReleaseSysCache(tuple); if the next step is an error?
+		// it is done in some places and omited in others.
+		ereport(ERROR,
```

Comment:
Is `ReleaseSysCache(tuple)` required before an error step? It is done in some cases but skipped in some others.